src/Microsoft.Performance.Testing/FakeAssembly.cs

```diff
@@ -12,23 +11,35 @@ public sealed class FakeAssembly
     {
         public FakeAssembly()
         {
-            this.TypesToReturn = Type.EmptyTypes;
+            this.AssemblyName = new AssemblyName("MyFakeAssembly");
             this.CodeBaseToReturn = Guid.NewGuid().ToString();
             this.LocationToReturn = Guid.NewGuid().ToString();
+            this.ReferencedAssemblies = Array.Empty<AssemblyName>();
+            this.TypesToReturn = Type.EmptyTypes;
         }
 
-        public Type[] TypesToReturn { get; set; }
+        public AssemblyName AssemblyName { get; set; }
         public string CodeBaseToReturn { get; set; }
         public string LocationToReturn { get; set; }
+        public AssemblyName[] ReferencedAssemblies { get; set; }
+        public Type[] TypesToReturn { get; set; }
 
         public override string CodeBase => this.CodeBaseToReturn;
         public override string Location => this.LocationToReturn;
 
-        public override Type[] GetTypes()
+        public override AssemblyName GetName()
         {
-            Assert.IsNotNull(this.TypesToReturn);
+            return this.AssemblyName;
+        }
 
-            return this.TypesToReturn;
+        public override AssemblyName[] GetReferencedAssemblies()
+        {
+            return this.ReferencedAssemblies ?? Array.Empty<AssemblyName>();
```

Comment:
>?? Array.Empty<AssemblyName>(); [](start = 45, length = 31)

why is this needed?

Reply:
asking for my own benefit. if it's not needed, i'm not suggestion you need to change it. just curious if there's a path I'm not thinking of.